src/overlays/actors/ovl_Bg_Mori_Elevator/z_bg_mori_elevator.c

```diff
@@ -20,37 +34,247 @@ const ActorInit Bg_Mori_Elevator_InitVars = {
     (ActorFunc)BgMoriElevator_Update,
     NULL,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A1800.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Bg_Mori_Elevator/func_808A18FC.s")
+static InitChainEntry sInitChain[] = {
+    ICHAIN_F32(uncullZoneForward, 2000, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneScale, 500, ICHAIN_CONTINUE),
+    ICHAIN_F32(uncullZoneDownward, 3000, ICHAIN_CONTINUE),
+    ICHAIN_VEC3F_DIV1000(scale, 1000, ICHAIN_STOP),
+};
+
+extern UNK_TYPE D_060035F8;
+extern Gfx D_06002AD0[];
+
+f32 func_808A1800(f32* posY, f32 target, f32 arg2, f32 yVel, f32 arg4) {
```

Comment:
Imo the arguments and overall style of this function should be adjusted based on the very similar functions in `z_lib.c` (such as `Math_SmoothScaleMaxMinF`)